int32 allied_type
---
float32 balance_ratio
int32 error